### DISCUSSION TITLE
Is there a lua API to dynamically set the window width?

### DISCUSSION THREAD

**Discussion by linrongbin16** [nvim-neo-tree/neo-tree.nvim#1047] [Q&A] [answered]
Hi, I want to adjust neo-tree window width by my screen monitor's size, e.g. I always use fullscreen of Neovim, but when using GUI client, the `&columns` value is going to change after GUI client finished its launch.

I'm going to listen to `VimResized` event, and if there's a lua API (for example `require('neotree').update_window_width(integer)`) that allow me to invoke and adjust neo-tree window width, so I can config neo-tree more dynamically fits into all kinds of screen monitor width.

**Comment by axelhj** [comment, marked as answer]
Hi, I did something related [here](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1250#issuecomment-1839113145). The API to resize neotree is just the Neovim window resize command, `vim.api.nvim_win_set_width(tree_win_handle, neotree_width)`.

The issue is finding the window handle which takes some search code. The issue comes when supporting multiple Neovim-tabs. HTH!

**Reply by linrongbin16** [reply]
thank you! it looks great, I will give it a try!
